src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -181,14 +181,14 @@ private static void EndInvokeDotNetAfterTask(Task task, JSRuntime jsRuntime, in
             }
         }
 
-        internal static object?[] ParseArguments(JSRuntime jsRuntime, string methodIdentifier, string arguments, Type[] parameterTypes)
+        internal static object?[] ParseArguments(JSRuntime jsRuntime, string methodIdentifier, SerializedArgs serializedArgs, Type[] parameterTypes)
         {
             if (parameterTypes.Length == 0)
             {
                 return Array.Empty<object>();
             }
 
-            var utf8JsonBytes = Encoding.UTF8.GetBytes(arguments);
+            var utf8JsonBytes = Encoding.UTF8.GetBytes(serializedArgs.ArgsJson ?? string.Empty);
```

Comment:
Is there some additional scenario where this might be `null` now but wouldn't have been before? Or could we avoid this by changing the type declaration of `SerializedArgs` to say that `ArgsJson` is non-null?